shelley/chain-and-ledger/executable-spec/test/Generator/Delegation.hs

```diff
@@ -76,12 +91,12 @@ genDCert
   -> DState
   -> Gen (Maybe (DCert, KeyPair))
 genDCert keys dState =
-  -- TODO @uroboros Generate _DeregKey_ Certificates
   -- TODO @uroboros Generate _RegPool_ Certificates
   -- TODO @uroboros Generate _RetirePool_ Certificates
   -- TODO @uroboros Generate _Delegate_ Certificates
-  Gen.frequency [ (25, genRegKeyCert keys dState)
-                , (75, pure Nothing)
+  Gen.frequency [ (1, genRegKeyCert keys dState)
+                , (1, genDeRegKeyCert keys dState)
+                , (1, pure Nothing)
```

Comment:
I'm calling this out as deserving consideration, since this is significantly different distribution from how it was before. With this weighting, we see in the test-suite:

```
    Classify Traces
      Ledger trace covers the relevant cases:                                OK (25.00s)
          ✓ Ledger trace covers the relevant cases passed 500 tests.
            there is at least 1 certificate for every 5 transactions          100% ████████████████████ ✓ 75%
            there is at least 1 RegKey certificate for every 5 transactions   100% ████████████████████ ✓ 75%
            there is at least 1 DeRegKey certificate for every 5 transactions  99% ███████████████████▊ ✓ 75%
            at most 75% of transactions have no certificates                  100% ████████████████████ ✓ 25%
```

***

with this weighting
```
  Gen.frequency [ (2, genRegKeyCert keys dState)
                , (2, genDeRegKeyCert keys dState)
                , (3, pure Nothing)
                ]
```

we see

```
    Classify Traces
      Ledger trace covers the relevant cases:                                OK (25.37s)
          ✓ Ledger trace covers the relevant cases passed 500 tests.
            there is at least 1 certificate for every 5 transactions          100% ████████████████████ ✓ 75%
            there is at least 1 RegKey certificate for every 5 transactions    98% ███████████████████▋ ✓ 75%
            there is at least 1 DeRegKey certificate for every 5 transactions  84% ████████████████▋··· ✓ 75%
            at most 75% of transactions have no certificates                  100% ████████████████████ ✓ 25%
```

***

and with this weighting 
```
  Gen.frequency [ (1, genRegKeyCert keys dState)
                , (1, genDeRegKeyCert keys dState)
                , (2, pure Nothing)
                ]
```

we see:
```
    Classify Traces
      Ledger trace covers the relevant cases:                                FAIL (23.80s)
          ✗ Ledger trace covers the relevant cases failed
            after 500 tests.
            there is at least 1 certificate for every 5 transactions          100% ████████████████████ ✓ 75%
            there is at least 1 RegKey certificate for every 5 transactions    88% █████████████████▋·· ✓ 75%
          ⚠ there is at least 1 DeRegKey certificate for every 5 transactions  54% ██████████▊········· ✗ 75%
            at most 75% of transactions have no certificates                  100% ████████████████████ ✓ 25%
```

Reply:
So, we could play with the weightings here. As we add more pairs to the `Gen.frequence` input list, we might also have to adjust down the thresholds [here](https://github.com/input-output-hk/cardano-ledger-specs/pull/922/files#diff-51eb3b7aef9dfc7364ae4b421310e558R38).